File: n11-client/src/services/wishlist.service.ts
import { http } from "./http";
import type { Page } from "@/types/api";

const enc = encodeURIComponent;

export type AdminWishlistEntry = {
  authId: number;
  userName: string;
  email: string | null;
  itemCount: number;
  productIds: string[];
};

export const wishlistService = {
  list: () => http.get<string[]>("/wishlist"),
  add: (productId: string) => http.post<string[]>(`/wishlist/add/${enc(productId)}`),
  remove: (productId: string) => http.delete<string[]>(`/wishlist/remove/${enc(productId)}`),
  clear: () => http.delete<void>("/wishlist/clear"),
  /** Admin: tüm kullanıcıların wishlist özeti. */
  adminListAll: (page = 0, size = 20) =>
    http.get<Page<AdminWishlistEntry>>("/wishlist/admin/find-all", {
      query: { page, size },
    }),
};

File: user-service/src/main/java/com/caglar/user/service/WishlistService.java
package com.caglar.user.service;

import com.caglar.user.dto.response.AdminWishlistEntryDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface WishlistService {
    List<String> getList(Long authId);
    List<String> add(Long authId, String productId);
    List<String> remove(Long authId, String productId);
    void clear(Long authId);

    /** Admin paneli — tüm kullanıcıların wishlist özetlerini sayfalı döner. */
    Page<AdminWishlistEntryDto> adminListAll(Pageable pageable);
}
